clients/include/testing_gels.hpp

```diff
@@ -35,6 +36,100 @@ inline void testname_gels(const Arguments& arg, std::string& name)
     hipblasGelsModel{}.test_name(arg, name);
 }
 
+template <typename T>
+inline hipblasStatus_t testing_gels_bad_arg(const Arguments& arg)
+{
+    auto hipblasGelsFn = arg.fortran ? hipblasGels<T, true> : hipblasGels<T, false>;
+
+    hipblasLocalHandle       handle(arg);
+    const int                M     = 100;
+    const int                N     = 101;
+    const int                nrhs  = 10;
+    const int                lda   = 102;
+    const int                ldb   = 103;
+    const hipblasOperation_t opN   = HIPBLAS_OP_N;
+    const hipblasOperation_t opBad = is_complex<T> ? HIPBLAS_OP_T : HIPBLAS_OP_C;
+
+    const int A_size = lda * N;
+    const int B_size = ldb * nrhs;
+
+    device_vector<T>   dA(A_size);
+    device_vector<T>   dB(B_size);
+    device_vector<int> dInfo(1);
+    int                info = 0;
+
+    EXPECT_HIPBLAS_STATUS(hipblasGelsFn(handle, opN, M, N, nrhs, dA, lda, dB, ldb, nullptr, dInfo),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGelsFn(handle, opBad, M, N, nrhs, dA, lda, dB, ldb, &info, dInfo),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-1, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGelsFn(handle, opN, -1, N, nrhs, dA, lda, dB, ldb, &info, dInfo),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-2, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGelsFn(handle, opN, M, -1, nrhs, dA, lda, dB, ldb, &info, dInfo),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-3, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGelsFn(handle, opN, M, N, -1, dA, lda, dB, ldb, &info, dInfo),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-4, info);
+
+    EXPECT_HIPBLAS_STATUS(
+        hipblasGelsFn(handle, opN, M, N, nrhs, nullptr, lda, dB, ldb, &info, dInfo),
+        HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-5, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGelsFn(handle, opN, M, N, nrhs, dA, M - 1, dB, ldb, &info, dInfo),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-6, info);
+
+    EXPECT_HIPBLAS_STATUS(
+        hipblasGelsFn(handle, opN, M, N, nrhs, dA, lda, nullptr, ldb, &info, dInfo),
+        HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-7, info);
+
+    EXPECT_HIPBLAS_STATUS(
+        hipblasGelsFn(handle, opN, 100, 200, nrhs, dA, lda, dB, 199, &info, dInfo),
+        HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-8, info);
+
+    EXPECT_HIPBLAS_STATUS(
+        hipblasGelsFn(handle, opN, 200, 100, nrhs, dA, 201, dB, 199, &info, dInfo),
+        HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-8, info);
```

Comment:
The size values are arbitrary to check correct return value when ldb < M or ldb < N. Thought it would be ok to be explicit in these cases.
The EXPECT_EQ values all correspond to the index of the argument which has the invalid value. In this case the 8th argument (not counting handle) is ldb, which is invalid as it is less than M.

Reply:
LOL okay if it is that simple but is this documented in their API, I only read the doc on Gels function and didn't see any mention of negative info values.

Reply:
Can you update the hipBLAS documentation to say this, and will see if solver team chimes in.  I see cublas has a function level comment: If info<0, the parameter in position -info is invalid.   That should be repeated for each function where it is true IMO

Reply:
hipBLAS mentions it in [hipblas.h documentation](https://github.com/ROCmSoftwarePlatform/hipBLAS/blob/develop/library/include/hipblas.h#L16818), although I suppose the wording is technically incorrect, I'll fix it in another commit. [cublas documentation](https://docs.nvidia.com/cuda/cublas/index.html#cublas-lt-t-gt-gelsBatched) as a reference.

Reply:
Thanks, I was reading rocsolver doc.   Okay maybe the fix is only needed in rocsolver doc, the j-th argument maybe clear enough is it 0 based indexing but handle is not referred to.

Reply:
I'm not really a fan of these explicit values, though it's more of a gut feeling than a logical objection. The approach to `lda` on line 85 is more to my liking. For `ldb`, perhaps we could instead do `max(M, N) - 1`.